### PR TITLE
Cleanup and fix in upgrade.php

### PR DESCRIPTION
**Pull request by stefaweb**
```
MailWatch for MailScanner Database Upgrade to 1.2.0 - RC4

Testing connectivity to the database ................................. OK

Updating database schema:

 - Convert database to utf8........................................... ALREADY DONE

 - Drop `geoip_country` table......................................... ALREADY DROPPED
 - Drop `spamscores` table............................................ ALREADY DROPPED
 - Add autorelease table to `mailscanner` database.................... ALREADY EXIST
 - Add resetid, resetexpire and lastreset fields in `users` table..... ALREADY EXIST

 - Fix schema for username field in `audit_log` table................. OK
 - Fix schema for id field in `blacklist` table....................... OK
 - Fix schema for username field in `users` table..................... OK
 - Fix schema for spamscore field in `users` table.................... OK
 - Fix schema for highspamscore field in `users` table................ OK
 - Fix schema for noscan field in `users` table....................... OK
 - Fix schema for username field in `user_filters` table.............. OK
 - Fix schema for username field in `whitelist` table................. OK
 - Fix schema for username field in `audit_log` table................. OK
 - Fix schema for password field in `users` table..................... OK
 - Fix schema for fullname field in `users` table..................... OK
 - Fix schema for rule_desc field in `mcp_rules` table................ OK

 - Add maillog_id field and primary key to `audit_log` table.......... ALREADY DONE
 - Add inq_id field and primary key to `inq` table.................... ALREADY DONE
 - Add maillog_id field and primary key to `maillog` table............ ALREADY DONE
 - Add mtalog_id field and primary key to `mtalog` table.............. ALREADY DONE
 - Add mtalog_id field and primary key to `outq` table................ ALREADY DONE
 - Add id field and primary key to `saved_filters` table.............. ALREADY DONE
 - Add mtalog_id field and primary key to `user_filters` table........ ALREADY DONE

 - Convert database to utf8mb4........................................ ALREADY DONE

 - Convert table `audit_log` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `autorelease` to utf8mb4............................. ALREADY CONVERTED
 - Convert table `blacklist` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `inq` to utf8mb4..................................... ALREADY CONVERTED
 - Convert table `maillog` to utf8mb4................................. ALREADY CONVERTED
 - Convert table `mcp_rules` to utf8mb4............................... ALREADY CONVERTED
 - Convert table `mtalog` to utf8mb4.................................. ALREADY CONVERTED
 - Convert table `mtalog_ids` to utf8mb4.............................. ALREADY CONVERTED
 - Convert table `outq` to utf8mb4.................................... ALREADY CONVERTED
 - Convert table `saved_filters` to utf8mb4........................... ALREADY CONVERTED
 - Convert table `sa_rules` to utf8mb4................................ ALREADY CONVERTED
 - Convert table `users` to utf8mb4................................... ALREADY CONVERTED
 - Convert table `user_filters` to utf8mb4............................ ALREADY CONVERTED
 - Convert table `whitelist` to utf8mb4............................... ALREADY CONVERTED

 - Convert table `audit_log` to InnoDB................................ ALREADY CONVERTED
 - Convert table `autorelease` to InnoDB.............................. ALREADY CONVERTED
 - Convert table `blacklist` to InnoDB................................ ALREADY CONVERTED
 - Convert table `inq` to InnoDB...................................... ALREADY CONVERTED
 - Convert table `maillog` to InnoDB.................................. ALREADY CONVERTED
 - Convert table `mcp_rules` to InnoDB................................ ALREADY CONVERTED
 - Convert table `mtalog` to InnoDB................................... ALREADY CONVERTED
 - Convert table `mtalog_ids` to InnoDB............................... ALREADY CONVERTED
 - Convert table `outq` to InnoDB..................................... ALREADY CONVERTED
 - Convert table `saved_filters` to InnoDB............................ ALREADY CONVERTED
 - Convert table `sa_rules` to InnoDB................................. ALREADY CONVERTED
 - Convert table `users` to InnoDB.................................... ALREADY CONVERTED
 - Convert table `user_filters` to InnoDB............................. ALREADY CONVERTED
 - Convert table `whitelist` to InnoDB................................ ALREADY CONVERTED

 - Search for missing indexes......................................... DONE

Checking MailScanner.conf settings:

 - QuarantineWholeMessage ............................................ OK
 - QuarantineWholeMessagesAsQueueFiles ............................... OK
 - DetailedSpamReport ................................................ OK
 - IncludeScoresInSpamAssassinReport ................................. OK
 - SpamActions ....................................................... OK
 - HighScoringSpamActions ............................................ WARNING
 - AlwaysLookedUpLast ................................................ OK

Checking conf.php configuration entry:

 - All needed entries are OK
 - All obsolete entries are already removed

*** ERROR/WARNING SUMMARY ***
MailScanner.conf: HighScoringSpamActions != store (=delete)
```

### CODE REVIEW

File: upgrade.php [hunk 1]
```diff
@@ -257,19 +270,7 @@ function getTableIndexes($table)
     // Table whitelist
     echo pad(' - Fix schema for username field in `whitelist` table');
     $sql = 'ALTER TABLE `whitelist` CHANGE `id` `id` bigint(11) UNSIGNED NOT NULL AUTO_INCREMENT';
```
Comment (endelwar) [inline upgrade.php]: please change this to bigint(20)

Comment (stefaweb) [inline upgrade.php]: ??
Wrong copy/paste somewhere.

**Comment by stefaweb** [comment]
How to correct this?
(add more .......)

```
 - Add autorelease table to `mailscanner` database.................... ALREADY EXIST
 - Add resetid, resetexpire and lastreset fields in `users` table............ ALREADY EXIST
```

**Comment by endelwar** [comment]
> How to correct this?

what's wrong with this output?

**Comment by stefaweb** [comment]
> what's wrong with this output?

Not aligned, look top of page.

**Comment by asuweb** [comment]
Seriously....this is an upgrade script, run once for each version - I really don't think we need to be concerned about the format.  No end user are ever going to see it, only the sysadmin the one time he / she runs the script

**Comment by stefaweb** [comment]
I no longer have the problem!
My server had to be in a bad mood.

**Comment by stefaweb** [comment]
> Seriously....

In old age one becomes maniac.  😀

**Pull request merged**